### PR TITLE
Feature/loan list v2

### PR DESCRIPTION
**Pull request by sinejespersen**
#### Link to issue

https://jira.itkdev.dk/browse/CMN-459

#### Description

FBS module
Publizon module
Loan list

#### Screenshot of the result

image

image

image

#### Checklist

- [ ] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [ ] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [ ] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

If you have any further comments or questions for the reviewer them please add them here.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -72,8 +72,6 @@ services:
       lagoon.type: cli-persistent
       lagoon.persistent.name: nginx # mount the persistent storage of nginx into this container
       lagoon.persistent: /app/web/sites/default/files/ # location where the persistent storage should be mounted
-    environment:
-      << : *default-environment # loads the defined environment variables from the top
```
Comment (cableman) [inline docker-compose.yml:76]: Why this change in the docker setup... don't think it should be altered.

File: web/modules/custom/dpl_loans/dpl_loans.info.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+name: DPL loans
+description: Handle the user profile loan pages
+package: DPL
+dependencies:
+  - drupal:block
+  - dpl_user_profile:dpl_user_profile
+  - dpl_fbs:dpl_fbs
+  - dpl_publizon:dpl_publizon
```
Comment (cableman) [inline web/modules/custom/dpl_loans/dpl_loans.info.yml:8]: Should this have a dependency to `dpl_react`?

File: web/modules/custom/dpl_loans/dpl_loans.module [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+/**
+ * @file
+ * Dpl_loans drupal module file.
+ *
+ * Demonstrates how to use the dpl_react module
+ * and render components.
+ */
```
Comment (cableman) [inline web/modules/custom/dpl_loans/dpl_loans.module:9]: Maybe update this block with correct comment :-)

File: web/modules/custom/dpl_loans/src/Form/LoanListSettingsForm.php [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    if (!filter_var($feesUrl, FILTER_VALIDATE_URL)) {
+      $form_state->setErrorByName('fees_page_url', $this->t('The url "%url" is not a valid URL.', ['%url' => $feesUrl]));
+    }
+
+    $materialUrl = $form_state->getValue('material_overdue_url');
+    if (!filter_var($materialUrl, FILTER_VALIDATE_URL)) {
+      $form_state->setErrorByName('material_overdue_url', $this->t('The url "%url" is not a valid URL.', ['%url' => $materialUrl]));
+    }
+
+    $pageSizeMobile = intval($form_state->getValue('page_size_mobile'));
```
Comment (cableman) [inline web/modules/custom/dpl_loans/src/Form/LoanListSettingsForm.php:82]: `intval` might not be the best way to do this, as intval will try to convert the value to an integer and hence you are not validating the actually value that gets stored.

See https://www.php.net/manual/en/function.intval.php 

A better solution would be:
```
$pageSizeMobile =  $form_state->getValue('page_size_mobile')
if (is_interger($pageSizeMobile) && $pageSizeMobile <= 0`) { ... } 
```
See https://www.php.net/manual/en/function.is-integer.php

File: web/modules/custom/dpl_loans/src/Form/LoanListSettingsForm.php [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    if ($pageSizeMobile <= 0) {
+      $form_state->setErrorByName('page_size_mobile', $this->t('Page size mobile has to be a positive integer'));
+    }
+
+    $pageSizeDesktop = intval($form_state->getValue('page_size_desktop'));
```
Comment (cableman) [inline web/modules/custom/dpl_loans/src/Form/LoanListSettingsForm.php:87]: Same here with intval

**Review by cableman — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (sinejespersen) [reply docker-compose.yml:76]: Did not notice when i reviewed the files, I think this is a leftover from Ture. I'll revert

File: config/sync/config_ignore.settings.yml [hunk 6]
```diff
@@ -97,5 +97,6 @@ ignored_config_entities:
   - ~user.settings
   - ~varnish_purger.settings.65fc931232
   - ~varnish_purger.settings.30cd45a1b1
+  - loan-list.settings
```
[the remark this answers is not in the document]
Comment (sinejespersen) [reply config/sync/config_ignore.settings.yml:100]: Should it start with tilde?

File: web/modules/custom/dpl_loans/dpl_loans.info.yml (hunk 2 above)
Comment (sinejespersen) [reply web/modules/custom/dpl_loans/dpl_loans.info.yml:8]: I dont know, should we?

File: web/modules/custom/dpl_loans/dpl_loans.module (hunk 3 above)
Comment (sinejespersen) [reply web/modules/custom/dpl_loans/dpl_loans.module:9]: good catch, will do

File: web/modules/custom/dpl_loans/src/Form/LoanListSettingsForm.php (hunk 4 above)
Comment (sinejespersen) [reply web/modules/custom/dpl_loans/src/Form/LoanListSettingsForm.php:82]: Thanks for this feedback, I'll implement

File: web/modules/custom/dpl_loans/src/Form/LoanListSettingsForm.php (hunk 5 above)
Comment (sinejespersen) [reply web/modules/custom/dpl_loans/src/Form/LoanListSettingsForm.php:87]: -||- ^

File: docker-compose.yml (hunk 1 above)
Comment (sinejespersen) [reply docker-compose.yml:76]: Checked it out, it seems "environment" is repeated 
https://github.com/danskernesdigitalebibliotek/dpl-cms/blob/develop/docker-compose.yml#L66-L76

File: web/modules/custom/dpl_loans/dpl_loans.info.yml (hunk 2 above)
Comment (sinejespersen) [reply web/modules/custom/dpl_loans/dpl_loans.info.yml:8]: I've created a dependency on dpl_react, that must be the way to go
